source/transactions-convenient-api/transactions-convenient-api.md

```diff
@@ -0,0 +1,384 @@
+`withTransaction` method will inspect the ClientSession's transaction state after the callback returns and take the most
+sensible course of action; however, it will not detect whether the callback has started a new transaction.
+
+### The callback function may be executed multiple times
+
+The implementation of withTransaction is based on the original examples for
+[Retry Transactions and Commit Operation](https://www.mongodb.com/docs/manual/core/transactions/#retry-transaction-and-commit-operation)
+from the MongoDB Manual. As such, the callback may be executed any number of times. Drivers are free to encourage their
+users to design idempotent callbacks.
+
+### The commit is retried after a write concern timeout (i.e. wtimeout) error
+
+Per the Transactions specification, drivers internally retry `commitTransaction` once if it fails due to a retryable
+error (as defined in the [Retryable Writes](../retryable-writes/retryable-writes.md#terms) specification). Beyond that,
+applications may manually retry `commitTransaction` if it fails with any error bearing the
+[UnknownTransactionCommitResult](../transactions/transactions.md#unknowntransactioncommitresult) error label. This label
+is applied for the the following errors:
+
+- Server selection failure
+- Retryable error (as defined in the [Retryable Writes](../retryable-writes/retryable-writes.md#terms) specification)
+- Write concern failure or timeout (excluding UnsatisfiableWriteConcern and UnknownReplWriteConcern)
+- MaxTimeMSExpired errors, ie `{ok:0, code: 50, codeName: "MaxTimeMSExpired"}` and
+  `{ok:1, writeConcernError: {code: 50, codeName: "MaxTimeMSExpired"}}`.
+
+A previous design for `withTransaction` retried for all of these errors *except* for write concern timeouts, so as not
+to exceed the user's original intention for `wtimeout`. The current design of this specification no longer excludes
+write concern timeouts, and simply retries `commitTransaction` within its timeout period for all errors bearing the
+"UnknownTransactionCommitResult" label.
+
+This change was made in light of the forthcoming Client-side Operations Timeout specification (see:
+[Future Work](#future-work)), which we expect will allow the current 120-second timeout for `withTransaction` to be
+customized and also obviate the need for users to specify `wtimeout`.
+
+### The commit is not retried after a MaxTimeMSExpired error
+
+This specification intentionally chooses not to retry commit operations after a MaxTimeMSExpired error as doing so would
+exceed the user's original intention for `maxTimeMS`.
+
+### The transaction and commit may be retried any number of times within a timeout period
+
+The implementation of withTransaction is based on the original examples for
+[Retry Transactions and Commit Operation](https://www.mongodb.com/docs/manual/core/transactions/#retry-transaction-and-commit-operation)
```

Comment:
Same feedback as I left for "The callback function may be executed multiple times" above. This is a dead anchor.

This paragraph could be reduced to:

> The transaction and commit may be continually retried as long as the error label
indicates that retrying is possible.

Reply:
Done